src/libraries/oob.proj

```diff
@@ -81,4 +78,12 @@
     <Message Condition="Exists('$(ArtifactsDir)packages')" Importance="High" Text="##vso[task.setvariable variable=_librariesBuildProducedPackages]true" />
   </Target>
 
+  <Import Project="Sdk.targets" Sdk="Microsoft.Build.NoTargets" />
+
+  <!-- Must be imported AFTER liveBuilds.targets (imported by Directory.Build.targets) which defines the ToolsILLinkDir. -->
+  <ImportGroup Condition="'$(BuildTargetFramework)' == '$(NetCoreAppCurrent)' or '$(BuildTargetFramework)' == ''">
+    <!-- Import the illink file which contains some of the logic required to illink the out-of-band assemblies. -->
+    <Import Project="$(RepositoryEngineeringDir)illink.targets" />
+  </ImportGroup>
```

Comment:
nit: would you mind reverting these changes here and instead define another property in the Directory.Build.targets where illink.targets is imported: https://github.com/dotnet/runtime/blob/7cc9f149bf4cc481f96e4c2e856908fc49f4adf8/src/libraries/Directory.Build.targets#L130

`Condition="'$(IsSourceProject)' == 'true' or '$(ExplicitlyImportCustomILLinkTargets)' == 'true'"